pydantic/_internal/_model_construction.py

```diff
@@ -60,7 +60,21 @@ def __setitem__(self, k: str, v: object) -> None:
         return super().__setitem__(k, v)
 
 
-@dataclass_transform(kw_only_default=True, field_specifiers=(PydanticModelField, PydanticModelPrivateAttr))
+def NonInitField(
+    default: Any = PydanticUndefined,
+    *,
+    default_factory: typing.Callable[[], Any] | None = None,
+    init: Literal[False] = False,
+) -> Any:
+    """Only for typing purpose. Used as default value of `__pydantic_fields_set__`,
+    `__pydantic_extra__`, `__pydantic_private__`, so they could be ignored when
+    synthesizing the `__init__` signature.
+    """
```

Comment:
Great idea. Makes things more explicit than `_PrivateAttr`. Maybe we could rename it `NoInitField`?